drv/lpc55-update-server/src/main.rs

```diff
@@ -503,6 +508,12 @@ impl idl::InOrderUpdateImpl for ServerImpl<'_> {
         ringbuf_entry!(Trace::State(self.state));
         self.next_block = None;
         self.fw_cache.fill(0);
+        // The sequence: [update, set transient preference, update] is legal.
+        // Clear any stale transient preference before update.
+        // Stage0 doesn't support transient override.
+        if component == RotComponent::Hubris {
+            set_hubris_transient_override(None);
+        }
```

Comment:
I think I see what this is trying to do but I'm not sure this is the right place. This would make `[set transient preference, update]` quietly not work as expected. I'm also not sure about the example `[update, set transient preference, update]` being legal but not supported. Can we clarify what the semantics of transient boot preference are supposed to be? Having it be effective for exactly one boot and clearing it on task startup seems like the least confusing semantics.

Reply:
Okay I re-read the RFD and double checked the bootleby source, the transient command is supposed to be nuked on boot by bootleby. I'm still not quite sure what makes this a 'stale' choice.

Reply:
It's not actually stale until some flash has been altered. Given the sequence where SlotId::B is active:
  - prep_image_update SlotId::A
  -  flash to hubris-a with version 1
  - switch_default_hubris_image(SlotId::A, SwitchDuration::Once)
  - prep_image_update SlotId::A
  - // abandon or abort the update before flashing anything
  - // reset the RoT
What image is preferred to boot?
 
This is a corner case since we normally expect update, set-preference, reset.
I'm thinking that in this case, since the RoT update server saw an intent to boot once into SlotId::A and then saw intent to update A again, that should cancel the boot once directive.

We could wait until the first block of an image is received or that block is written to cancel the boot once directive.

Reply:
I agree it's a corner case. Fixing it in this way is a pretty poor developer experience because there's no indication that the boot setting has changed which will frustrate everyone. The transient boot setting should not be cancelled without some indication. We also have the exact same issue with the persistent settings. I get the theory is to avoid booting into a potentially bad image because we're now relying on bootleby to correctly determine that the slot is unbootable but this feels like it causes more problems than it solves. I think this issue of boot settings would be better tracked as a separate issue.

Reply:
I think that an alternate approach will work; If current boot preference points to a flash bank that would be altered by some operation, then the operation fails. The remedy would be to abort the update or set the boot preference away from the flash bank to be altered.
So, Active and Alt start with valid images, Alt is then set as next boot preference, attempt to update Alt fails.
Setting boot preference to the Active image works, then updating Alt works, then setting preference to Alt works.
Appropriate errors are generated on the failures as feedback.

Reply:
I think that's a very solid approach